svm-interface/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+//! The interface of the Solana SVM.
+//!
+//! This crate outlines the Solana SVM specification in Rust.
+//! Developers who wish to build custom SVM implementations in Rust can use
+//! this interface to build a specification-compliant SVM, complete with an
+//! integration testing harness to ensure maximum compatibility with the
+//! network.
+
+pub mod load_results;
+pub mod results;
+
+use {
+    load_results::TransactionLoadResult, results::TransactionExecutionResult,
+    solana_sdk::transaction::SanitizedTransaction,
+};
+
+/// The main interface for the Solana SVM.
+///
+/// At its core, the SVM is a transaction batch processor.
+/// Given a batch of Solana transactions, any SVM implementation must be able
+/// to process the batch and return a result.
+/// This functionality is fully extentable, allowing developers to add custom
+/// functionality, as well as configure their own monitoring tooling, such as
+/// timings, metrics, and logging.
+pub trait TransactionBatchProcessorInterface<C, T>
```

Comment:
why not just use `SVMInterface` instead of the longer `TransactionBatchProcessorInterface`?

Reply:
Ohh yeah I'm not married to any particular naming conventions here. In fact, I would even opt for `TransactionBatchProcessor` as the trait and `AgaveTransactionBatchProcessor` as the implementation, but I thought this might reduce the diff (it didn't matter, diff is huge anyway).

Reply:
If, as you correctly commented, _At its core, the SVM is a transaction batch processor_, it seems the interface could be named simply as `SVM`, could it not?

Reply:
Yep, it could!